app/core/services/chat_history.py

```
@@ -0,0 +1,40 @@
+        key = f"chat:{session_id}"
+        items = r.lrange(key, -(CHAT_HISTORY_MAX_TURNS * 2), -1)
+        history = [json.loads(item) for item in items]
+
+        print(f"[ChatHistory] 조회: session_id={session_id}, key={key}, count={len(history)}")
+        for i, msg in enumerate(history):
+            role = msg.get("role", "?")
+            content = msg.get("content", "")[:50]  # 50자까지만 출력
+            print(f"  [{i}] {role}: {content}{'...' if len(msg.get('content', '')) > 50 else ''}")
+
+        return history
+    except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)